File: apps/android/app/src/main/java/com/futo/notes/CrashlogEndpoint.kt
package com.futo.notes

import android.content.SharedPreferences

object CrashlogEndpoint {
    private const val PRODUCTION_BASE_URL = "https://notes-crashlog.futo.org"
    private const val STAGING_BASE_URL = "https://staging-notes-crashlog.futo.org"
    private const val LOCAL_BASE_URL = "http://10.0.2.2:5100"

    @Volatile private var prefs: SharedPreferences? = null

    fun attach(prefs: SharedPreferences) {
        this.prefs = prefs
    }

    fun devBaseUrl(useStaging: Boolean): String =
        if (useStaging) STAGING_BASE_URL else LOCAL_BASE_URL

    val baseUrl: String
        get() {
            if (!BuildConfig.DEBUG) return PRODUCTION_BASE_URL
            return devBaseUrl(prefs?.getBoolean(Prefs.CRASHLOG_STAGING, false) == true)
        }
}

File: apps/ios/Sources/CrashReporting/CrashlogEndpoint.swift
import Foundation

enum CrashlogEndpoint {
    static let stagingKey = "futo.crashlog.useStaging"

    private static let productionBaseUrl = "https://notes-crashlog.futo.org"
    private static let stagingBaseUrl = "https://staging-notes-crashlog.futo.org"
    private static let localBaseUrl = "http://localhost:5100"

    static func devBaseUrl(useStaging: Bool) -> String {
        useStaging ? stagingBaseUrl : localBaseUrl
    }

    static var baseUrl: String {
        #if DEBUG
            devBaseUrl(useStaging: UserDefaults.standard.bool(forKey: stagingKey))
        #else
            productionBaseUrl
        #endif
    }

    static func url(_ path: String) -> URL {
        URL(string: baseUrl + path)!
    }
}
